st2api/st2api/controllers/v1/actionexecutions.py

```diff
@@ -369,7 +389,15 @@ def format(gen):
             return gen
 
         def make_response():
-            app_iter = itertools.chain(existing_output_iter(), new_output_iter())
+            if existing_only:
+                # Only return existing output produced by now and return immediately. Don't wait
+                # for execution to finish
+                print('laaaaa')
+                print(existing_only)
+                app_iter = existing_output_iter()
+            else:
+                app_iter = itertools.chain(existing_output_iter(), new_output_iter())
+
             res = Response(content_type='text/plain', app_iter=app_iter)
```

Comment:
Eventsource also needs certain headers: `Cache-Control: no-cache` and `Content-Type: text/event-stream`

Reply:
optionally, we can also look for `Last-Event-ID` request header and only output results after a certain timestamp.